epub33/core/index.html

```diff
@@ -436,14 +453,8 @@ <h3>Terminology</h3>
 						<dfn id="dfn-path-name" data-lt="Path Names">Path Name</dfn>
 					</dt>
 					<dd>
-						<p>For a given directory within the <a href="#sec-container-abstract">OCF Abstract
-							Container</a>, the string holding all directory <a>File Name</a> in the full path
-							concatenated together with a <code>/</code> (<code>U+002F</code>) character separating the
-							directory File Names.</p>
-						<p>For a given file within the OCF Abstract Container, the Path Name is the string holding all
-							directory File Names concatenated together with a <code>/</code> character separating the
-							directory File Names, followed by a <code>/</code> character and then the File Name of the
-							file.</p>
+						<p>The Path Name of a file or directory <var>file</var> is the <a data-cite="url#concept-url-path">path</a> of the <a>content URL</a> for <var>file</var>. 
+						It is derived from the <a>File Name</a> of <var>file</var> following the steps specified in <a href="#sec-file-names-to-path-names"></a>.</p>
```

Comment:
That definition doesn't work:
For a given file, the result "Deriving the Path Names of Files" algorithm does not necessarily equals the path of its content URL. This is the case when the file names have characters that are illegal in URL.

For instance, consider a file "my doc.xhtml" in the "OPS" subdirectory of the root directory. It's Path Name is `"OPS/my doc.xhtml"` but the serialized path component of its URL is `"OPS/my%20doc.xhtml`.

I would say something along the lines of:

(if we opt for an axiomatic version):
> The Path Name of a file or directory is its full path relative to the root directory, where path components are separated by the `U+002F (/)` character.

(if we opt for an algorithmic version):
> The Path Name of a file or directory is its full path relative to the root directory, as defined by the "Deriving the Path Names of Files" algorithm.

Reply:
> For a given file, the result "Deriving the Path Names of Files" algorithm does not necessarily equals the path of its content URL. This is the case when the file names have characters that are illegal in URL.

Oops, yes.

I realized that the fact of turning, in your example, the space character into %20 is in the definition of the content URL in the new URL section: as far as I see the official URL parsing algorithm does that. (I hope that is correct. Because if it is not that we have to say somewhere that the file names are _converted_ when prepending into path not just copied.)

However... I have no problem with your reformulation of the definition, and I will do it in the next commit. I will stick with the algorithmic version for now; again, I would like to get this PR merged as soon as we all agree, because solving the core issue is what is really important at this point, and push the algorithmic/declarative question aside for a while.

Reply:
> as far as I see the official URL parsing algorithm does that. (I hope that is correct. Because if it is not that we have to say somewhere that the file names are _converted_ when prepending into path not just copied.) 

That's how I understand it as well 😊, by virtue of step 2.3 of _path state_. Note that if a character is not allowed in URLs, it will raise a validation error along the way (step 2.1 of _path state_) but still continue to step 2.3 to produce a valid result.

> Is it worth making a note about this in the new section, just to make this clear to everyone reading the spec? Reading the official URL spec is not everyone's cup of tea…

That's what I suggested in [a previous comment](https://github.com/w3c/epub-specs/pull/1898#discussion_r746645782) 😉 

> However... I have no problem with your reformulation of the definition, and I will do it in the next commit. I will stick with the algorithmic version for now; again, I would like to get this PR merged as soon as we all agree, because solving the core issue is what is really important at this point, and push the algorithmic/declarative question aside for a while.

OK, works for me!

Reply:
See https://github.com/w3c/epub-specs/pull/1898#issuecomment-966899712 for further discussion, and resolving this particular thread.